### PR TITLE
Port System.IO triple slash to docs

### PR DESCRIPTION
**Pull request by carlossanlop**
Used DocsPortingTool and also added some missing descriptions manually.

### CODE REVIEW

File: xml/System.IO/FileStreamOptions.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
   </Base>
   <Interfaces />
   <Docs>
-    <summary>To be added.</summary>
+    <summary>Defines a variety of configuration options for <see. cref="T:System.IO.FileStream"/>.</summary>
```
Comment (carlossanlop) [inline xml/System.IO/FileStreamOptions.xml:17]: @adamsitnik @Jozkee does this description sound good to you? The class was not documented in triple slash.

File: xml/System.IO/RandomAccess.xml [hunk 2]
```diff
@@ -14,8 +14,12 @@
   </Base>
   <Interfaces />
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Provides offset-based APIs for reading and writing files, ensuring atomic and thread-safe operations.</summary>
```
Comment (carlossanlop) [inline xml/System.IO/RandomAccess.xml:17]: @adamsitnik This class didn't have a description, so I took some inspiration from the first API proposal you wrote:

https://github.com/dotnet/runtime/issues/24847#issuecomment-807050146

And also from the PR description:

https://github.com/dotnet/runtime/pull/53669

Do you agree with the provided information?

File: xml/System.IO/FileStreamOptions.xml (hunk 1 above)
Comment (adamsitnik) [reply xml/System.IO/FileStreamOptions.xml:17]: Sound good to me 👍

File: xml/System.IO/StreamReader.xml [hunk 3]
```diff
@@ -553,10 +553,24 @@
...
-        <summary>To be added.</summary>
+        <param name="path">The complete file path to be read.</param>
+        <param name="options">An object that specifies the configuration options for the underlying <see cref="T:System.IO.FileStream"/>.</param>
+        <summary>Initializes a new instance of the <see cref="T:System.IO.StreamReader" /> class for the specified file path, using the default encoding, enabling detection of byte order marks at the beginning of the file, and configured with the specified <see cref="T:System.IO.FileStreamOptions"/> object.</summary>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException"><paramref name="stream" /> is not readable.
+
+-or-
+
+          <paramref name="path" /> is an empty string ("").</exception>
```
Comment (adamsitnik) [inline xml/System.IO/StreamReader.xml:564]: nit: extra empty lines

```suggestion
-or-
          <paramref name="path" /> is an empty string ("").</exception>
```

File: xml/System.IO/StreamReader.xml [hunk 4]
```diff
@@ -1040,12 +1054,25 @@
...
+        <param name="path">The complete file path to be read.</param>
+        <param name="encoding">The character encoding to use.</param>
+        <param name="detectEncodingFromByteOrderMarks">Indicates whether to look for byte order marks at the beginning of the file.</param>
+        <param name="options">An object that specifies the configuration options for the underlying <see cref="T:System.IO.FileStream"/>.</param>
+        <summary>Initializes a new instance of the <see cref="T:System.IO.StreamReader" /> class for the specified file path, with the specified character encoding, byte order mark detection option, and configured with the specified <see cref="T:System.IO.FileStreamOptions"/> object.</summary>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException"><paramref name="stream" /> is not readable.
+
+          -or-
+
```
Comment (adamsitnik) [inline xml/System.IO/StreamReader.xml:1066]: nit: extra empty lines

```suggestion
          -or-
```

File: xml/System.IO/RandomAccess.xml (hunk 2 above)
Comment (adamsitnik) [reply xml/System.IO/RandomAccess.xml:17]: I was initailly using the "atomic" word in the issue description, as I meant lack of two separate operations: seek to given offset and read|write from current offset. But after I've learned about the existence of [CreateFileTransacted](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createfiletransacteda) and some of the file system complexities (the fact that write has succeeded does not mean that data is going to be stored on the disk) I believe that it's better to not use this word. Thread-safe is all we need.

Sorry for not adding that in the first place when I was adding `RandomAccess` type.

```suggestion
    <summary>Provides offset-based APIs for reading and writing files in a thread-safe manner.</summary>
```

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you for doing this @carlossanlop !

File: xml/System.IO/StreamReader.xml (hunk 3 above)
Comment (carlossanlop) [reply xml/System.IO/StreamReader.xml:564]: Those are needed here to ensure they render properly in html. It's in triple slash where we can remove them to save code lines.

File: xml/System.IO/StreamReader.xml (hunk 4 above)
Comment (carlossanlop) [reply xml/System.IO/StreamReader.xml:1066]: Same as above.

File: xml/System.IO/StreamReader.xml (hunk 3 above)
Comment (adamsitnik) [reply xml/System.IO/StreamReader.xml:564]: @carlossanlop thanks, I was not aware of that!

**Pull request merged**